presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1968,7 +1970,15 @@ else if (partitionUpdate.getUpdateMode() == NEW || partitionUpdate.getUpdateMode
                         partitionUpdate.getStatistics(),
                         columnTypes,
                         getColumnStatistics(partitionComputedStatistics, partition.getValues()));
-                metastore.addPartition(session, handle.getSchemaName(), handle.getTableName(), partition, partitionUpdate.getWritePath(), partitionStatistics);
+                metastore.addPartition(
+                        session,
+                        handle.getSchemaName(),
+                        handle.getTableName(),
+                        partition,
+                        partitionUpdate.getWritePath(),
+                        partitionStatistics,
+                        false,
+                        handle.getLocationHandle().getTargetPath());
```

Comment:
It might be cleaner and more readable to extract the table location directly from the table object